### PR TITLE
[User History] Improvements

### PR DESCRIPTION
**Pull request by mkangia**
## Summary

Did a full scan of changes in https://github.com/dimagi/commcare-hq/pull/29887 that revealed scope of improvement.
Best review by commit

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story

Mostly documentation. Tests present for minor change to role logging.

### Rollback instructions

same as base PR

### CODE REVIEW

File: corehq/apps/cleanup/tasks.py [hunk 1]
```diff
@@ -245,18 +239,6 @@ def check_for_ucr_tables_without_existing_domain():
...
-@periodic_task(run_every=crontab(minute=0, hour=16), queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
-def delete_web_user():
-    if settings.SERVER_ENVIRONMENT == 'production':
-        for username in [
-            'create_growth' + '@' + 'outlook.com',
-            'growth_analytics' + '@' + 'outlook.com',
-        ]:
-            web_user = WebUser.get_by_username(username)
-            if web_user:
-                web_user.delete(None, deleted_by=SYSTEM_USER_ID, deleted_via=__name__ + '.delete_web_user')
```
Comment (mkangia) [inline corehq/apps/cleanup/tasks.py:257]: confirmed good to remove https://github.com/dimagi/commcare-hq/pull/29901#issuecomment-865830527

File: corehq/apps/api/tests/core_api.py [hunk 2]
```diff
@@ -398,7 +398,7 @@ def setUp(self):
                                                  None, None)
 
     def tearDown(self):
-        self.commcare_user.delete(self.domain, deleted_by=None)
+        self.commcare_user.delete(self.domain.name, deleted_by=None)
```
Comment (orangejenny) [inline corehq/apps/api/tests/core_api.py:401]: I guess the `domain_obj` convention hasn't made it all the way throughout HQ.

File: corehq/apps/cleanup/tasks.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cleanup/tasks.py:257]: 🏆  for following through on this, thank you

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/api/tests/core_api.py (hunk 2 above)
Comment (mkangia) [reply corehq/apps/api/tests/core_api.py:401]: There were few of these instances, mostly in tests iirc

**Pull request merged**